pysb/simulator/ssa_base.py

```diff
@@ -0,0 +1,159 @@
+                      if isinstance(sym, Expression)}:
+                rate = re.sub(r'\b%s\b' % e.name,
+                              expr_strings[e.name],
+                              rate)
+
+            matches = re.findall(pattern, rate)
+            for m in matches:
+                repl = m[0]
+                for i in range(1, int(m[1])):
+                    repl += "*(%s-%d)" % (m[0], i)
+                rate = re.sub(pattern, repl, rate)
+
+            rate = re.sub(r'_*s(\d+)',
+                          lambda m: 'y[%s]' % (int(m.group(1))),
+                          rate)
+            for q, prm in enumerate(params_names):
+                rate = re.sub(r'\b(%s)\b' % prm, 'param_vec[%s]' % q, rate)
+            items = rate.split('*')
+
+            # create rate string. Places param_vec upfront to ensure that
+            # the resulting value is a double (type of param_vec) and not
+            # degraded to an int
+            rate = [i for i in sorted(items) if i.startswith('param_vec')]
+            rate += [i for i in sorted(items) if not i.startswith('param_vec')]
+            rate = "*".join(rate)
+            rate = re.sub('\*$', '', rate)
+            rate = re.sub('d0', '', rate)
+            rate = p.sub('', rate)
+            rate = rate.replace('pow', 'powf')
+            hazards_string += rate + ";\n"
+        return dict(n_species=self._n_species, n_params=self._parameter_number,
+                    n_reactions=_reaction_number, hazards=hazards_string,
+                    stoch=stoich_string)
+
+    def run(self, tspan=None, param_values=None, initials=None, number_sim=0):
+
+        num_sim = int(number_sim)
+
+        # check for proper arguments
+        if param_values is None and initials is None and not num_sim:
+            raise SimulatorException("Please provide a multi-dimension set of "
+                                     "parameters, initials, or number_sim>0")
+        elif param_values is None and not num_sim:
+            num_sim = np.array(initials).shape[0]
+        elif initials is None and not num_sim:
+            num_sim = np.array(param_values).shape[0]
+
+        if param_values is None:
+            # Run simulation using same param_values
+            param_values = np.repeat(self.param_values, num_sim, axis=0)
+        elif len(param_values.shape) == 1:
+            param_values = np.repeat([param_values], num_sim, axis=0)
+
+        if initials is None:
+            # Run simulation using same initial conditions
+            initials = np.repeat(self.initials, num_sim, axis=0)
+        elif len(initials.shape) == 1:
+            initials = np.repeat([initials], num_sim, axis=0)
+        self.num_sim = num_sim
+        super(SSABase, self).run(tspan=tspan, initials=initials,
+                                 param_values=param_values,
+                                 _run_kwargs=locals())
+
+    @staticmethod
+    def _get_stoch(model):
```

Comment:
Small point, but should this be `_get_stoich`?

Reply:
Good catch!